sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventDataBatch.cs

```diff
@@ -212,7 +212,7 @@ internal void Clear()
         ///
         /// <returns>The set of events as an enumerable of the requested type.</returns>
         ///
-        internal IEnumerable<T> AsEnumerable<T>() => InnerBatch.AsEnumerable<T>();
+        internal List<T> AsList<T>() => InnerBatch.AsList<T>();
```

Comment:
Would it be worth aligning now to Azure Service Bus that uses AsReadOnly?

Reply:
Once I figure out #9569, the batch will be caching `AmqpMessage` rather than `EventData` to avoid the cost of serializing multiple times, and we'll lost type symmetry anyway.  That said, there's harm in making it `ReadOnlyCollection` since I'm using that in the message converter for symmetry.